src/plugins/analysis/hardware_analysis/code/hardware_analysis.py

```diff
@@ -0,0 +1,115 @@
+import lief
+
+from analysis.PluginBase import AnalysisBasePlugin
+
+
+class AnalysisPlugin(AnalysisBasePlugin):
+    '''
+    Hardware Analysis Plug-in
+    '''
+    NAME = 'hardware_analysis'
+    DESCRIPTION = 'Hardware Analysis Plug-in'
+    DEPENDENCIES = ['cpu_architecture', 'elf_analysis', 'kernel_config']
+    VERSION = '0.2.1'
+
+    def __init__(self, plugin_adminstrator, config=None, recursive=True):
+        '''
+        recursive flag: If True recursively analyze included files
+        '''
+        self.config = config
+
+        super().__init__(plugin_adminstrator, config=config, recursive=recursive, plugin_path=__file__)
+
+    def process_object(self, file_object):
+
+        # init the data
+
+        cpu_architecture_dict = file_object.processed_analysis['cpu_architecture']
+        kernel_config_dict = file_object.processed_analysis['kernel_config']
+
+        # getting the cpu architecture (test needed?)
+        cpu_architecture = list(cpu_architecture_dict.keys())[0]
+
+        if cpu_architecture == "summary":
+            cpu_architecture = "no information"
```

Comment:
it would be better to store `None` here and display `"no information"` in the template